From 6cad949f6fe99a1b7f4e425ad448a0a43bd6206e Mon Sep 17 00:00:00 2001
From: Li Wang <li.wang@windriver.com>
Date: Wed, 26 Aug 2015 10:30:57 +0800
Subject: [PATCH 56/68] fs: vmfs: Use generic mmap function

the patch comes from:
git://git.yoctoproject.org/linux-yocto-3.10
commit 7f08192ff737c0b3ebc88c017711e539daf5a91b upstream

Use the generic mmap function to avoid possible deadlock on vmfs_mutex.

Signed-off-by: Anders Berg <anders.berg@intel.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
Signed-off-by: Li Wang <li.wang@windriver.com>
---
 fs/vmfs/file.c |   23 +----------------------
 1 files changed, 1 insertions(+), 22 deletions(-)

diff --git a/fs/vmfs/file.c b/fs/vmfs/file.c
index 18a403f..8814b40 100644
--- a/fs/vmfs/file.c
+++ b/fs/vmfs/file.c
@@ -253,27 +253,6 @@ out:
 	return status;
 }
 
-static int vmfs_file_mmap(struct file *file, struct vm_area_struct *vma)
-{
-	struct dentry *dentry = file->f_path.dentry;
-	int status;
-
-	VERBOSE("file %s/%s, address %lu - %lu\n",
-		DENTRY_PATH(dentry), vma->vm_start, vma->vm_end);
-
-	mutex_lock(&vmfs_mutex);
-	status = vmfs_revalidate_inode(dentry);
-	mutex_unlock(&vmfs_mutex);
-	if (status) {
-		PARANOIA("%s/%s validation failed, error=%d\n",
-			 DENTRY_PATH(dentry), status);
-		goto out;
-	}
-	status = generic_file_mmap(file, vma);
-out:
-	return status;
-}
-
 static ssize_t
 vmfs_file_splice_read(struct file *file, loff_t *ppos,
 		      struct pipe_inode_info *pipe, size_t count,
@@ -486,7 +465,7 @@ const struct file_operations vmfs_file_operations = {
 	.write = do_sync_write,
 	.aio_write = vmfs_file_aio_write,
 	.unlocked_ioctl = vmfs_unlocked_ioctl,
-	.mmap = vmfs_file_mmap,
+	.mmap = generic_file_mmap,
 	.open = vmfs_file_open,
 	.release = vmfs_file_release,
 	.fsync = vmfs_fsync,
-- 
1.7.5.4

